connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -39,6 +39,63 @@ message Expression {
     UnresolvedRegex unresolved_regex = 8;
     SortOrder sort_order = 9;
     LambdaFunction lambda_function = 10;
+    Window window = 11;
+  }
+
+
+  // Expression for the OVER clause or WINDOW clause.
+  message Window {
+
+    // (Required) The window function.
+    Expression window_function = 1;
+
+    // (Optional) The way that input rows are partitioned.
+    repeated Expression partition_spec = 2;
+
+    // (Optional) Ordering of rows in a partition.
+    repeated SortOrder order_spec = 3;
+
+    // (Optional) Window frame in a partition.
+    //
+    // If not set, it will be treated as 'UnspecifiedFrame'.
+    WindowFrame frame_spec = 4;
```

Comment:
```suggestion
    optional WindowFrame frame_spec = 4;
```

Reply:
it seems we don't add `optional` for other proto messages:

for example
```
message Project {
  // (Optional) Input relation is optional for Project.
  //
  // For example, `SELECT ABS(-1)` is valid plan without an input plan.
  Relation input = 1;

  // (Required) A Project requires at least one expression.
  repeated Expression expressions = 3;
}
```